.github/osia_convert.rb

```diff
@@ -1,268 +1,365 @@
 require_relative 'osia_helper'
 require 'date'
 
-README = 'README.md'
-ARCHIVE = 'ARCHIVE.md'
-APPSTORE = 'APPSTORE.md'
-
-NOT_ENGLISH = '🌐'
-ARCHIVE_TAG = 'archive'
-
-def apps_archived(apps)
-  a = apps.select { |a| a['tags'] != nil }.select { |b| b['tags'].include?(ARCHIVE_TAG) }
-  a.sort_by { |k, v| k['title'].downcase }
-end
-
-def apps_for_cat(apps, id)
-  f = apps.select do |a|
-
-    tags = a['tags']
-    if tags.nil?
-      true
-    else
-      !(tags.include?(ARCHIVE_TAG))
-    end
+# File constants 
+README = "README.md"
+ARCHIVE = "ARCHIVE.md"
+APPSTORE = "APPSTORE.md"
+LATEST = "LATEST.md"
+
+# Shows that a project is archived. 
+ARCHIVE_TAG = "archive"
+
+LATEST_APPS_COUNT = 15
+
+##
+#
+class App
+  attr_reader :title
+  attr_reader :source
+  attr_reader :itunes
+  attr_reader :homepage
+  attr_reader :description
+  attr_reader :tags
+  attr_reader :stars
+  attr_reader :lang
+  attr_reader :date_added
+  attr_reader :screenshots
+  attr_reader :license
+
+  def initialize(json)
+     @title = json["title"]
+     @source = json["source"]
+     @itunes = json["itunes"]
+     @homepage = json["homepage"]
+     @description = json["description"]
+     @tags = json["tags"]
+     @stars = json["stars"]
+     @lang = json["lang"]
+     @date_added = json["date_added"]
+     @screenshots = json["screenshots"]
+     @license = json["license"]
   end
 
-  s = f.select do |a|
-    cat = a['category-ids']
-    cat.class == Array ? cat.include?(id) : (cat == id)
-  end
-  s.sort_by { |k, v| k['title'].downcase }
 end
 
-def app_store_total(j)
-  apps = j['projects']
-  s = apps.reject { |x| x['itunes'].nil? }
+##
+#
+def app_store_total(source)
+  apps = source["projects"]
+  apps = select_app_store_apps(apps)
```

Comment:
Something weird about using `apps` For line 49 and 50

Can this be improved by using `apps = select_app_store_apps(source["projects"])` ?